nemo/collections/asr/inference/pipelines/buffered_rnnt_pipeline.py

```diff
@@ -174,6 +175,99 @@ def init_decoding_computer(self) -> None:
+            )
+
+            if num_prompts_int > 0 and is_dict_like:
+                return {
+                    'num_prompts': num_prompts_int,
+                    'prompt_dict': prompt_dict,
+                    'compute_dtype': getattr(self.asr_model.asr_model, 'dtype', torch.float32),
+                }
+
+        return {}
+
+    def _resolve_prompt_index(self, language_code: str) -> int:
+        """
+        Resolve language_code to a strict prompt index; raise if invalid.
+        Args:
+            language_code: (str) Language code to resolve (e.g., "en-US", "es-ES").
+        Returns:
+            (int) Prompt index corresponding to the language code.
+        Raises:
+            RuntimeError: If prompt configuration is missing.
+            ValueError: If language_code is not found in prompt dictionary.
+        """
+        if not hasattr(self, '_prompt_config') or not self._prompt_config:
+            raise RuntimeError("Prompt configuration is missing for a prompt-enabled model.")
+        prompt_dict = self._prompt_config['prompt_dict']
+        lang_index = prompt_dict.get(language_code, None)
+        if lang_index is None:
+            raise ValueError(
+                f"Language code '{language_code}' not found in prompt dictionary. "
+                f"Available languages: {list(prompt_dict.keys())}"
+            )
+        return lang_index
+
+    def _get_prompt_matrix(self) -> Tensor:
+        """
+        Return cached identity matrix [num_prompts, num_prompts] on device/dtype.
+        Returns:
+            (Tensor) Identity matrix for prompt selection.
+        """
+        if not hasattr(self, '_prompt_config') or not self._prompt_config:
+            raise RuntimeError("Prompt configuration is missing for a prompt-enabled model.")
+        key = (self.device, self._prompt_config['compute_dtype'])
+        cached = self._prompt_matrix_cache.get(key)
+        if cached is not None:
+            return cached
+        num_prompts = self._prompt_config['num_prompts']
+        compute_dtype = self._prompt_config['compute_dtype']
+        eye = torch.eye(num_prompts, device=self.device, dtype=compute_dtype)
+        self._prompt_matrix_cache[key] = eye
+        return eye
+
+    def _build_prompt_vectors(self, states: list) -> Tensor:
+        """
+        Build prompt vectors for a batch of states.
+        Args:
+            states: (list) List of streaming states.
+        Returns:
+            (Tensor) Prompt vectors of shape [B, num_prompts].
+        Raises:
+            ValueError: If any prompt index is out of range.
+        """
+        indices = torch.tensor([getattr(s, 'prompt_idx', 0) for s in states], device=self.device, dtype=torch.long)
+        num_prompts = self._prompt_config['num_prompts']
+        if torch.any((indices < 0) | (indices >= num_prompts)):
+            raise ValueError("Found out-of-range prompt index in batch.")
+        prompt_matrix = self._get_prompt_matrix()
```

Comment:
If I correctly understand the logic, you need to construct a tensor [B, num_prompts] using one-hot vectors `indices`. In this case I suggest using `torch.one_hot(indices, num_classes=num_prompts)` and avoid using cache (simplify logic, the same efficiency, since `index_select` constructs a new tensor).